mautic/mautic
# Create mtc_id and mtc_sid cookies for Mautic public pages to be congruent with mtc.js

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Pages that leverage the tracking code create mtc_is and mtc_sid cookies on the domain the code is embedded. This makes it easy for analytics code to get the ID of the contact Mautic is tracking. But, for Mautic's own public pages such as landing pages, these cookies do not exist. This results in making it tricky to get the contact ID for analytics code setup in Mautic's Configuration. This PR creates those cookies to make this possible and remain congruent with 3rd party sites.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Visit a Mautic landing page as an anonymous contact and view the cookies
2. Note there are no mtc_id or mtc_sid session cookies

#### Steps to test this PR:
1. Apply PR, and repeat above
2. There should now be mtc_id and mtc_sid session cookies
